### PR TITLE
#15 ADD type definition to enable use with typescript

### PR DESCRIPTION
**Pull request by liamsorsby**

### CODE REVIEW

**Comment by astorm** [comment]
jenkins run the tests please

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (astorm) [inline tsconfig.json:1]: @liamsorsby could you talk a little about the intent behind this tsconfig.json file?  Is this just here to allow someone to use `tsc` to generate the `types.d.ts` file, or are there more reasons for having this here?

**Review by astorm — COMMENTED** [review]
@liamsorsby Thanks for the PR! I've got one review question for you but other than that is seems simple enough -- get that question answered and we'll accept/merge.

Also -- just to set expectations -- with OpenTracing having moved on to to OpenTelementry,  it was our understanding that this package was mostly a historical curiosity.  It's in a state of benign neglect. PRs accepted, but we're not actively maintaining it. 

Your interest says that might be incorrect -- if you're able to share we'd be curious to know where/how you're using  this.

File: tsconfig.json (hunk 1 above)
Comment (liamsorsby) [reply tsconfig.json:1]: @astorm Hey, yes this was just created to ensure that the config used to generate the types file was consistent. However, now I think about it, this should really be a step in the CI and pushed as part of the build tagging.

**Comment by liamsorsby** [comment]
@astorm I wasn't actually aware that you'd moved over to OpenTelementry. The reason I was looking at this repo was purely due to following the elastic documentation which now makes me think it might need updating: https://www.elastic.co/guide/en/apm/agent/nodejs/current/opentracing.html
Is the correct documentation https://www.elastic.co/guide/en/apm/get-started/current/open-telemetry-elastic.html

**Comment by astorm** [comment]
Yeah -- it's a confusing state of affairs.  The Elastic Node.js agen't hasn't exactly _moved_ to OpenTelemetry.  The timeline is more like this

1. The OpenTracing project starts up

2. We, The Elastic Node.js Agent, build the open tracing bridge that lets you create an open tracing tracer that sends spans to the Elastic agent, which sends data to the APM server which sends them on to an Elasticsearch/Kibana instance.  

3. The OpenTracing project merges with Open Census to form OpenTelementry, and replaces most of their APIs along the way 

4. We, Elastic APM (not just the Node.js agent), build an exporter for the _OpenTelemetry Collector_ which allows the OpenTelemetry Collector to send data to APM Server which sends data to an Elasticsearch/Kibana instance. (and you send data to the Open Telemetry Collector using the OpenTelemetry javascript instrumentations)

So, OpenTracing and OpenTelemetry are different projects (even though it's a lot of the same people).  So we keep this repo around for anyone still using OpenTracing.

TL;DR; -- yeah, our docs probably need to be updated to make that distinction clearer.  :) 

Regardless -- thanks for the PR.  We're going to merge since some types are better than no types.  If you have any other questions about all this don't hesitate to ask.

**Review by astorm — APPROVED** [review]
